pkg/endpoint/policy.go

```diff
@@ -416,14 +416,16 @@ func (e *Endpoint) regenerateL3Policy(repo *policy.Repository) (bool, error) {
 	return valid, err
 }
 
+// Note that this function assumes that endpoint policy has already been generated!
 // must be called with endpoint.Mutex held for reading
 func (e *Endpoint) updateNetworkPolicy(owner Owner, proxyWaitGroup *completion.WaitGroup) error {
-	// Skip updating the NetworkPolicy if no policy has been calculated.
+	// Skip updating the NetworkPolicy if no identity has been computed for this
+	// endpoint.
 	// This breaks a circular dependency between configuring NetworkPolicies in
 	// sidecar Envoy proxies and those proxies needing network connectivity
 	// to get their initial configuration, which is required for them to ACK
 	// the NetworkPolicies.
-	if !e.policyCalculated || e.SecurityIdentity == nil {
+	if e.SecurityIdentity == nil {
```

Comment:
Perhaps we can check `nextPolicyRevision` for the endpoint here.

Reply:
Since this function is called after policy is regenerated isn't also true that the endpoint must have an identity? or do we handle `e.SecurityIdentity == nil` as identity Init?

Reply:
`regeneratePolicy` exits early if the endpoint's `SecurityIdentity` field is nil. I'm not 100 % about whether we can be assured that the Endpoint's security identity is nil or non-nil because identity is assigned via a controller on a per-endpoint basis.